src/System.Collections/src/System/Collections/Generic/HashSet.cs

```diff
@@ -1711,6 +1716,95 @@ internal static bool HashSetEquals(HashSet<T> set1, HashSet<T> set2, IEqualityCo
         }
 
         /// <summary>
+        /// Internal method used for HashSetEqualityComparer. Compares set1 and set2 according 
+        /// to specified comparer.
+        /// Because items are hashed according to a specific equality comparer, we have to resort
+        /// to building a new set if neither use that equality comparer.
+        /// </summary>
+        /// <param name="set1">The first set to compare.</param>
+        /// <param name="set2">The second set to compare.</param>
+        /// <param name="comparer">The comparer for comparing elements.</param>
+        /// <returns><c>true</c> if the two sets are equivalent, <c>false</c> otherwise.</returns>
+        internal static bool HashSetEquals(HashSet<T> set1, HashSet<T> set2, IEqualityComparer<T> comparer)
+        {
+            if (set1 == set2)
+            {
+                return true;
+            }
+
+            if (set1 == null | set2 == null)
```

Comment:
Is Boolean logic really measurably faster here?  IIRC it results in more IL, and it looks like a mistake, so unless there's a measurable throughput improvement to using it, let's just use logical.